mintpy/view.py

```diff
@@ -956,12 +956,18 @@ def read_data4figure(i_start, i_end, inps, metadata):
     # mask
     if inps.msk is not None:
         vprint('masking data')
-        msk = np.tile(inps.msk, (data.shape[0], 1, 1))
-        data = np.ma.masked_where(msk == 0., data)
+        inps.msk = np.tile(inps.msk, (data.shape[0], 1, 1))
+        data = np.ma.masked_where(inps.msk == 0., data)
+
+    if inps.msk is None:
```

Comment:
This is the 3D input data scenario, where we don't want to save the 3D mask matrix.